draft-ietf-privacypass-architecture.md

```diff
@@ -107,162 +82,251 @@ of all participating entities.
+Privacy Pass is a protocol for authorization based on anonymous-credential
+authentication mechanisms. Typical approaches for authorizing clients,
+such as through the use of long-term cookies, are not privacy-friendly
+since they allow servers to track clients across sessions and interactions.
+Privacy Pass takes a different approach: instead of presenting linkable
+state carrying information to servers, e.g., whether or not the client
+is an authorized user or has completed some prior challenge, clients
+present unlinkable proofs that attest to this information.
+
+The most basic Privacy Pass protocol provides a set of cross-origin
+authorization tokens that protect the client's anonymity during interactions
+with a server. This allows clients to communicate an attestation of a
+previously authenticated server action, without having to reauthenticate
+manually. The tokens retain anonymity in the sense that the act of
+revealing them cannot be linked back to the session where they were
+initially issued.
+
+At a high level, Privacy Pass is composed of two protocols: issuance
+and redemption.
+
+The issuance protocol runs between a Client and two network functions in the
+Privacy Pass architecture: Attestation and Issuance. These two network
+functions can be implemented by the same protocol participant, but can also be
+implemented separately. The Issuer is responsible for issuing tokens in
+response to requests from Clients. The Attester is responsible for attesting
+properties about the Client for which tokens are issued. The Issuer needs to be
+trusted by the server that later redeems the token. Attestation can be
+performed by the Issuer or by an Attester that is trusted by the Issuer.
+Clients might prefer to select different Attesters, separate from the Issuer,
+to be able to use preferred authentication methods or improve privacy by not
+directly communicating with an Issuer. Depending on the attestation,
+Attesters can store state about a Client, such as the number of overall tokens
+issued thus far. As an example of an Issuance protocol, in the original Privacy
+Pass protocol {{PPSRV}}, tokens were only issued to Clients that solved
+CAPTCHAs. In this context, the Attester attested that some client solved a
+CAPTCHA and the resulting token produced by the Issuer was proof of this fact.
+
+The redemption protocol runs between Client and Origin (server). It allows
+Origins to challenge Clients to present one or more tokens for authorization.
+Depending on the type of token, e.g., whether or not it is cross-origin
+or per-origin, and whether or not it can be cached, the Client either presents
+a previously obtained token or invokes the issuance protocol to acquire one
+for authorization.
+
+The issuance and redemption protocols operate in concert as shown in
+the figure below.
 
-The following terms are used throughout this document.
+~~~
+ Client        Attester          Issuer          Origin
```

Comment:
Maybe worth moving Origin to the left to match the other figures?

Reply:
Sure, I'll play with it and see what works.